refactor masking strategy factory for improved extensibility (#560)
ethyca/fidesops: docs/fidesops/docs/guides/masking_strategies.md

## Extensibility

In fidesops, masking strategies are all built on top of an abstract base class - `MaskingStrategy`. 

> `MaskingStrategy` has five methods - `mask`, `secrets_required`,  `get_configuration_model`, `get_description`, and `data_type_supported`. For more detail on these 

Reviewer: There's an extra space here 🔍 
```suggestion
`MaskingStrategy` has five methods - `mask`, `secrets_required`, `get_configuration_model`, `get_description`, and `data_type_supported`. For more detail on these 
```

any defaults that should be applied in their absence. All configuration classes extend from the 
`MaskingConfiguration` class.

### Integrating with the Masking Strategy Factory

> In order to be able to leverage a masking strategy that's been implemented, the `MaskingStrategy` subclass needs to be registered with the `MaskingStrategyFactory`. Doing this is as simple as putting the `@MaskingStrategyFactory.register(<strategy_name>)` decorator on the `MaskingStrategy` subclass, as shown in the example the section above.

Reviewer: ```suggestion
In order to leverage an implemented masking strategy, the `MaskingStrategy` subclass must be registered with the `MaskingStrategyFactory`. To register a new `MaskingStrategy`, use the `register` decorator on the `MaskingStrategy` subclass definition, as shown in the above example.
```

> The value passed as the argument to the decorator must be the registered name of the `MaskingStrategy` subclass. This is the value that should be passed by callers in the `"masking_strategy"."strategy"` field, in order to invoke this masking strategy.

Reviewer 2: ```suggestion
The value passed as the argument to the decorator must be the registered name of the `MaskingStrategy` subclass. This is the same value defined by [callers](#using-fidesops-as-a-masking-service) in the `"masking_strategy"."strategy"` field.
```

> Make sure to pass in the registered name of the `MaskingStrategy` subclass as the argument to the decorator. This is the value that will need to be passed by callers in the `"masking_strategy"."strategy"` field in order to invoke this masking strategy, either as part of an erasure request or with a standalone call against the `/masking/mask` endpoint).

Reviewer: ```suggestion
The value passed as the argument to the decorator must be the registered name of the `MaskingStrategy` subclass. This is the value that should be passed by callers in the `"masking_strategy"."strategy"` field, in order to invoke this masking strategy.
```
